src/providers/structure.ts

```diff
@@ -40,8 +46,50 @@ export class SectionNodeProvider implements vscode.TreeDataProvider<Section> {
     }
 
     update() {
-        this.refresh()
-        this._onDidChangeTreeData.fire(undefined)
+        this.refresh().finally(() => {this._onDidChangeTreeData.fire(undefined)})
+    }
+
+    async parseBibTeX() {
```

Comment:
My 2c suggestion. Maybe renaming this function to something like `buildBibtexModel` would be more coherent with the `buildModel` method (which could become `buildLatexModel`). I guess it would make the symmetry clearer between computing structures for bibtex and latex even if the fact this function is `async` makes his calls a bit different.